### PR TITLE
OHM-54: Middleware convert js

### PR DESCRIPTION
**Pull request by rcrichton**
Closes: #350 

@devcritter another one for review.

### CODE REVIEW

File: src/middleware/visualizer.coffee [hunk 1]
```diff
@@ -85,9 +85,7 @@ storeVisualizerEvents = (ctx, done) ->
...
 
-exports.koaMiddleware =  `function *visualizerMiddleware(next) {
-  yield next;
-  if (config.visualizer.enableVisualizer) {
-    storeVisualizerEvents(this, function(){});
-  }
-}`
+exports.koaMiddleware = (next) ->
+  yield next
+  if config.visualizer.enableVisualizer?
```
Comment (devcritter) [inline src/middleware/visualizer.coffee]: I think you may need to take the question mark out, I suspect that'll just translate to `if (config.visualizer.enableVisualizer != null)`

File: src/middleware/tcpBypassAuthentication.coffee [hunk 2]
```diff
@@ -16,15 +16,11 @@ exports.authenticateUser = (ctx, done) ->
...
+  yield authenticateUser this
 
-  if (this.authenticated) {
-    yield next;
-  } else {
-    this.response.status = "unauthorized";
-  }
-  
-}`
+  if this.authenticated?
```
Comment (devcritter) [inline src/middleware/tcpBypassAuthentication.coffee]: Same as the below comment regarding the question mark

**Comment by devcritter** [comment]
Looks good to me! So much neater in coffeescript :) There's just a few places where you have `if variable?` but I believe that won't get interpreted as a boolean check, just nullity. Otherwise :+1:

**Comment by rcrichton** [comment]
Yeah it really is! Good catch on the visualiser boolean. The others are all `this.authenticated` which is actually an client object or null/undefined if not authenticated, so I think the existential operator is ok there.

**Comment by devcritter** [comment]
Aah, right. Cool, yeah

**Pull request merged**